src/PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.cpp

```diff
@@ -205,79 +205,176 @@ tnsr::aa<DataType, SpatialDim, Frame> compute_pi(
 }
 
 template <size_t SpatialDim, typename Frame, typename DataType>
-tnsr::a<DataType, SpatialDim, Frame> compute_spacetime_normal_one_form(
+tnsr::a<DataType, SpatialDim, Frame> spacetime_normal_one_form(
     const Scalar<DataType>& lapse) noexcept {
   auto normal_one_form =
-      make_with_value<tnsr::a<DataType, SpatialDim, Frame>>(get<>(lapse), 0.);
-  get<0>(normal_one_form) = -get<>(lapse);
+      make_with_value<tnsr::a<DataType, SpatialDim, Frame>>(lapse, 0.);
+  get<0>(normal_one_form) = -get(lapse);
   return normal_one_form;
 }
 
 template <size_t SpatialDim, typename Frame, typename DataType>
-tnsr::A<DataType, SpatialDim, Frame> compute_spacetime_normal_vector(
+tnsr::A<DataType, SpatialDim, Frame> spacetime_normal_vector(
     const Scalar<DataType>& lapse,
     const tnsr::I<DataType, SpatialDim, Frame>& shift) noexcept {
   auto spacetime_normal_vector{
-      make_with_value<tnsr::A<DataType, SpatialDim, Frame>>(get<>(lapse), 0.)};
-  get<0>(spacetime_normal_vector) = 1. / get<>(lapse);
+      make_with_value<tnsr::A<DataType, SpatialDim, Frame>>(lapse, 0.)};
+  get<0>(spacetime_normal_vector) = 1. / get(lapse);
   for (size_t i = 0; i < SpatialDim; i++) {
     spacetime_normal_vector.get(i + 1) =
         -shift.get(i) * get<0>(spacetime_normal_vector);
   }
   return spacetime_normal_vector;
 }
 
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> extrinsic_curvature(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::iJ<DataType, SpatialDim, Frame>& deriv_shift,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric,
+    const tnsr::ii<DataType, SpatialDim, Frame>& dt_spatial_metric,
+    const tnsr::ijj<DataType, SpatialDim, Frame>&
+        deriv_spatial_metric) noexcept {
+  const DataType half_over_lapse = 0.5 / get(lapse);
+
+  auto ex_curvature =
+      make_with_value<tnsr::ii<DataType, SpatialDim, Frame>>(lapse, 0.0);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = i; j < SpatialDim; ++j) {  // Symmetry
+      for (size_t k = 0; k < SpatialDim; ++k) {
+        ex_curvature.get(i, j) +=
+            shift.get(k) * deriv_spatial_metric.get(k, i, j) +
+            spatial_metric.get(k, i) * deriv_shift.get(j, k) +
+            spatial_metric.get(k, j) * deriv_shift.get(i, k);
+      }
+      ex_curvature.get(i, j) -= dt_spatial_metric.get(i, j);
+      ex_curvature.get(i, j) *= half_over_lapse;
+    }
+  }
+
+  return ex_curvature;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> gauge_source_function(
```

Comment:
Move into a different file for GH things

Move into the GeneralizedHarmonic namespace (this namespace exists already because of the RHS implementation for GH)